docs/python/sdk-guide.md

```diff
@@ -109,13 +109,19 @@ Otherwise, the JSON blob is a list of the individual results.
 >>> async def main():
 ...     async with Session() as sess:
 ...         client = OrdersClient(sess)
-...         orders = client.list_orders()
-...         orders_list = collect(orders)
+...         orders_aiter = client.list_orders_aiter()
+...         orders_list = collect(orders_aiter)
```

Comment:
@jreiberkyle @kevinlacaille let's not lead users to assign client.list_orders_aiter() to a variable. The best usage is just one single line.

```
orders = [order for order in client.list_orders()]
```

Lets recommend this and do it everywhere in this project.

I'm also :-1: on `_aiter`. There's no synchronous alternative and the fact that it's a generator is documented in the method docstring and in the type hint.

Reply:
A note about the history of "iter" in names in In Python: whenever we've seen "iter" used as a prefix on method names, it's been part of a transition. Once, Python dicts had `keys()`, and it was a list. But then it was decided to make the keys method a generator function. To help the transition, `iterkeys()` was added. It returned a generator while `keys()` returned a list. And then in Python 3, `keys()` returned a generator and `iterkeys()` was deprecated.